### ISSUE TITLE
[EKS] [request]: Managed Node Groups using role with extra path do not join cluster

### ISSUE THREAD

**Issue by dpiddockcmp** [aws/containers-roadmap#926]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
Bug report - Managed Node Group nodes are unable to join a cluster when using an IAM role with extra path.

**Which service(s) is this request for?**
EKS, Managed Node Groups

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard**
The MNG service adds a section to aws-auth's mapRoles with the full ARN for the role. But when joining the cluster aws-iam-authenticator does not receive the full ARN as `GetCallerIdentity` lacks the extra path element.

Adding a section to the ConfigMap with the "correct" ARN before creating the MNG results in a duplicate block. Also nicely demonstrates the two cases:

```
  mapRoles: |
    - groups:
      - system:bootstrappers
      - system:nodes
      rolearn: arn:aws:iam::111111111111:role/test-eks
      username: system:node:{{EC2PrivateDNSName}}
    - groups:
      - system:bootstrappers
      - system:nodes
      rolearn: arn:aws:iam::111111111111:role/extra/test-eks
      username: system:node:{{EC2PrivateDNSName}}
```

**Are you currently working around this issue?**
`aws-auth` ConfigMap must be edited to "correct" the IAM role ARN

**Additional context**
Related to:
- #573
- https://github.com/kubernetes-sigs/aws-iam-authenticator/issues/153
- https://github.com/terraform-aws-modules/terraform-aws-eks/issues/893 

**Attachments**

**Comment by anarsen** [comment]
This behavior is documented here:

- https://docs.aws.amazon.com/eks/latest/userguide/troubleshooting_iam.html#security-iam-troubleshoot-ConfigMap
- https://docs.aws.amazon.com/eks/latest/userguide/add-user-role.html (see 2.b)

**Comment by dpiddockcmp** [comment]
It would be good if the aws-auth inserts generated by AWS when using Managed Node Groups followed their own documentation 😆

**Comment by tsndqst** [comment]
@mikestef9 Any updates on this?  It's been almost two years and this is still an issue.

**Comment by portswigger-tim** [comment]
Anyone using CDK that ends up here, I did this as a "workaround": https://github.com/aws/containers-roadmap/issues/573#issuecomment-1378964224

**Comment by mikestef9** [comment]
Addressed with #185
